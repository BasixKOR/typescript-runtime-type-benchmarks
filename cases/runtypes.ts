import { Boolean, Number, String, Record } from 'runtypes';
import { Data } from '../data'

const DataType = Record({
  number: Number,
  neg_number: Number.withConstraint(n => n < 0),
  max_number: Number,
  string: String,
  long_string: String.withConstraint(s => s.length > 100),
  boolean: Boolean,
  deeplyNested: Record({
    foo: String,
    num: Number,
    bool: Boolean
  })
})

export function caseRuntypes(data: Data) {
  return DataType.check(data)
}
