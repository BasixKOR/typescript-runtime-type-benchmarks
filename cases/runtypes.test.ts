import { caseRuntypes } from './runtypes'
import { DATA } from '../data'

describe('runtypes', () => {
  it('should pass', () => {
    const res = caseRuntypes(DATA)
    expect(res).toEqual(DATA)

    if (res) {
      expect(res.number).toEqual(DATA.number)
    }
  })

  it('should fail negative number test', () => {
    const req = Object.assign({}, DATA, { neg_number: 1 })
    expect(() => caseRuntypes(req)).toThrowErrorMatchingSnapshot()
  })

  it('should fail min length test', () => {
    const req = Object.assign({}, DATA, { long_string: '' })
    expect(() => caseRuntypes(req)).toThrowErrorMatchingSnapshot()
  })
})
