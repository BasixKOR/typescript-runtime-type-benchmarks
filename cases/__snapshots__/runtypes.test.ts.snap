// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`runtypes should fail min length test 1`] = `"Failed constraint check"`;

exports[`runtypes should fail negative number test 1`] = `"Failed constraint check"`;
